### PR TITLE
[#201] Additional checks 360: check for grant title length < 140 chars

### PR DESCRIPTION
**Pull request by edugomez**

### CODE REVIEW

File: cove_360/lib/threesixtygiving.py [hunk 1]
```diff
@@ -396,6 +396,18 @@ def process(self, grant, grant_flat, path_prefix):
...
 
+class TitleLength(AdditionalTest):
+    def process(self, grant, grant_flat, path_prefix):
+        title = grant.get("title", '')
+        if len(title) > 140:
+            self.failed = True
+            self.count += 1
+            self.json_locations.append(path_prefix + '/title')
+
+        self.heading = self.format_heading_count("a title longer that recommended")
```
Comment (robredpath) [inline cove_360/lib/threesixtygiving.py]: longer than*

File: cove_360/tests.py [hunk 2]
```diff
@@ -352,12 +353,16 @@
...
-    ({'heading': "2 grants have a title and a description that are the same",
+    ({'heading': "1 grant has a title and a description that are the same",
       'message': ("Users may find that the data is less useful as they are unable to "
                   "discover more about the grants. Consider including a more detailed "
                   "description if you have one.")},
-     ['grants/1/description', 'grants/2/description'],
-     [['grants', 'Z', 3, 'Description'], ['grants', 'Z', 4, 'Description']]),
+     ['grants/2/description'],
+     [['grants', 'Z', 4, 'Description']]),
+    ({'heading': "1 grant has a title longer that recommended",
```
Comment (robredpath) [inline cove_360/tests.py]: longer than*

File: cove_360/lib/threesixtygiving.py [hunk 3]
```diff
@@ -396,6 +396,18 @@ def process(self, grant, grant_flat, path_prefix):
         self.message = "Users may find that the data is less useful as they are unable to discover more about the grants. Consider including a more detailed description if you have one."
 
 
+class TitleLength(AdditionalTest):
+    def process(self, grant, grant_flat, path_prefix):
+        title = grant.get("title", '')
```
Comment (robredpath) [inline cove_360/lib/threesixtygiving.py]: How does this handle the case of (invalid) data that doesn't have a title field?

Comment (robredpath) [inline cove_360/lib/threesixtygiving.py]: To be clear: in the current state, we don't expect Additional Checks to produce meaningful results for invalid data. But I do want to check that it doesn't break horribly (eg throw a 500) if the data is invalid.

File: cove_360/lib/threesixtygiving.py (hunk 1 above)
Comment (edugomez) [inline cove_360/lib/threesixtygiving.py]: Will change that

File: cove_360/lib/threesixtygiving.py (hunk 3 above)
Comment (edugomez) [inline cove_360/lib/threesixtygiving.py]: If there's no title field this test won't produce any output (`grant.get('title', '')` will avoid any `KeyError` raising a 500 in the server if there's no title field)

**Pull request merged**